File: src/minimal_example/src/subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include "minimal_example/GetCounter.h"

static int g_counter = 0;

void chatterCallback(const std_msgs::String::ConstPtr& msg) {
    g_counter++;
}

bool getCounter(minimal_example::GetCounter::Request  &req,
                minimal_example::GetCounter::Response &res) {
    res.counter = g_counter;
    return true;
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "listener");
    ros::NodeHandle n;
    ros::Subscriber sub = n.subscribe("chatter", 10, chatterCallback);
    ros::ServiceServer service = n.advertiseService("counter", getCounter);
    ros::spin();
    return 0;
}

File: src/minimal_python/src/minimal_python/subscriber.py
# -*- coding: utf-8 -*-

# SPDX-License-Identifier: MIT
# Copyright © 2021 André Santos

import rospy
from std_msgs.msg import String
from minimal_example.srv import GetCounter, GetCounterResponse

g_counter = 0

def chatter_callback(msg):
    global g_counter
    g_counter += 1

def get_counter(req):
    return GetCounterResponse(g_counter)

def main(args):
    try:
        rospy.init_node('listener')
        sub = rospy.Subscriber('chatter', String, chatter_callback,
                               queue_size=10)
        service = rospy.Service('counter', GetCounter, get_counter)
        rospy.spin()
    except rospy.ROSInterruptException:
        pass
    return 0

